File: app/src/main/java/org/jellyfin/androidtv/ui/presentation/ChannelCardPresenter.java
package org.jellyfin.androidtv.ui.presentation;

import android.view.View;
import android.view.ViewGroup;

import androidx.leanback.widget.Presenter;

import org.jellyfin.androidtv.ui.card.ChannelCardView;
import org.jellyfin.apiclient.model.livetv.ChannelInfoDto;

public class ChannelCardPresenter extends Presenter {
    class ViewHolder extends Presenter.ViewHolder {
        private ChannelCardView mCardView;

        public ViewHolder(View view) {
            super(view);
            mCardView = (ChannelCardView) view;
        }

        public void setItem(ChannelInfoDto item) {
            mCardView.setItem(item);
        }
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent) {
        ChannelCardView view = new ChannelCardView(parent.getContext());

        view.setFocusable(true);
        view.setFocusableInTouchMode(true);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(Presenter.ViewHolder viewHolder, Object item) {
        if (!(item instanceof ChannelInfoDto)) return;
        ChannelInfoDto channel = (ChannelInfoDto) item;

        ViewHolder vh = (ViewHolder) viewHolder;

        vh.setItem(channel);
    }

    @Override
    public void onUnbindViewHolder(Presenter.ViewHolder viewHolder) {
        // Unused
    }
}

File: app/src/main/java/org/jellyfin/androidtv/ui/presentation/ChannelCardPresenter.kt
package org.jellyfin.androidtv.ui.presentation

import android.view.ViewGroup
import androidx.leanback.widget.Presenter
import org.jellyfin.androidtv.ui.card.ChannelCardView
import org.jellyfin.sdk.model.api.BaseItemDto

class ChannelCardPresenter : Presenter() {
	class ViewHolder(
		private val cardView: ChannelCardView,
	) : Presenter.ViewHolder(cardView) {
		fun setItem(item: BaseItemDto?) = cardView.setItem(item)
	}

	override fun onCreateViewHolder(parent: ViewGroup): ViewHolder {
		val view = ChannelCardView(parent.context).apply {
			isFocusable = true
			isFocusableInTouchMode = true
		}

		return ViewHolder(view)
	}

	override fun onBindViewHolder(viewHolder: Presenter.ViewHolder, item: Any?) {
		if (item !is BaseItemDto) return
		if (viewHolder !is ViewHolder) return

		viewHolder.setItem(item)
	}

	override fun onUnbindViewHolder(viewHolder: Presenter.ViewHolder) = Unit
}
